charts/dependency-track/templates/tls-secret.yaml

```diff
@@ -0,0 +1,30 @@
+{{- if and (eq (.Values.tls.enabled | toString) "true") (.Values.tls.existingSecret | empty) }}
```

Comment:
Thank you for pointing out the use of `extraDeploy` and the common practice of manual configuration of cert-manager in Kubernetes deployments. I appreciate your perspective and understand that for skilled IT specialists like yourself, the prevalence of manual configuration of cert-manager is quite familiar.

However, our decision to maintain the current configuration separately and explicitly is intentional. For a broader range of users, including those who may not be as familiar with Kubernetes intricacies, the process must be straightforward. Time is very precious, and learning to manually configure settings like these can be quite time-consuming and potentially error-prone for those less experienced with Kubernetes' more advanced configurations. Additionally, `extraDeploy` is designed for advanced users or specific edge cases, not for everyday use.

We aim to keep our Helm chart as accessible as possible, enabling users to achieve a secure deployment with minimal additional research or modification to the default settings. This approach reduces the initial barrier for new users and speeds up the process of deploying secure, production-ready instances.

I believe that maintaining this feature as part of our default configuration supports our goal of user-friendliness and broad accessibility. However, I'm open to further discussion and exploring ways we could offer more advanced configurations as optional, without compromising the ease of use for new users. Your insights are very valuable, and I look forward to continuing this conversation to find the best path forward.

Reply:
This template extends beyond a basic self-signed certificate, representing a sophisticated mechanism that integrates seamlessly with the Ingress Template. While its primary utility is to facilitate the creation of a self-signed certificate, this is but one aspect of its functionality. For users who rely on their own or an external CA, and do not use Cert Manager, the template provides an easy method to include a certificate without requiring in-depth technical knowledge.

Your current assessment seems to misunderstand the template’s full scope. The functionality of extraResource (formerly known as extraDeploy—please note the name change) is distinct and designed as an advanced tool for specialized scenarios.

Additionally, I would like to clarify that the template is user-friendly. The notion that it is difficult to manage or that a deep understanding of Kubernetes is essential for deploying Helm charts is not accurate. The template also does not restrict advanced users as suggested; it is designed to simplify the process for both novices and experienced professionals alike.

It might be more beneficial to focus on enhancing quality and conducting unit tests rather than engaging in discussions on what might be considered minor points. While I appreciate the importance of thorough documentation, as reflected in the values.yaml and my contributions, it is important to recognize that not all administrators actively seek detailed technical documentation.

Finally, the assertion that adding toggle options complicates the chart, while stating that learning basic configurations is simple, seems contradictory. No offense is intended by these observations.

I hope we can redirect our focus to constructive discussions that enhance the tool’s usability and effectiveness.

Reply:
My short answer is, let's agree to disagree. 

I think readability is key to be able to maintain software in the long run, no matter what your needs might be.
I created https://github.com/DependencyTrack/helm-charts/pull/23 to showcase how you can solve the same issue, which I personally think is a better option and I know you won't agree on that :).

I will use this helm chart no matter which PR gets merged, as long as the issue gets solved. 

FYI, I don't think disagreeing is an issue, we just see things differently and that is fine.